spec/win32/unit/certstore_spec.rb

```diff
@@ -186,6 +186,97 @@
     end
   end
 
+  CERT_SYSTEM_STORE_LOCAL_MACHINE = 0x00020000
+
+  describe "#get_cert_pfx" do
+    context "When passing empty certificate store name" do
+      let(:store_name) { "" }
+      it "raises ArgumentError" do
+        expect { certstore.open(store_name) }.to raise_error(ArgumentError, "Empty Certificate Store.")
+      end
+    end
+
+    context "When passing empty thumbprint" do
+      let(:store_name) { "root" }
+      let(:thumbprint) { " " }
+      it "raises ArgumentError" do
+        store = certstore.open(store_name)
+        expect { store.get(thumbprint) }.to raise_error(ArgumentError, "Invalid certificate thumbprint.")
+      end
+    end
+
+    context "When passing thumbprint is nil" do
+      let(:store_name) { "root" }
+      let(:thumbprint) { nil }
+      it "raises ArgumentError" do
+        store = certstore.open(store_name)
+        expect { store.get(thumbprint) }.to raise_error(ArgumentError, "Invalid certificate thumbprint.")
+      end
+    end
+
+    context "When passing invalid thumbprint" do
+      let(:store_name) { "root" }
+      let(:thumbprint) { "b1bc968bd4f49d622aa89a81f2150152a41d829c" }
+      let(:password) { "1234" }
+      before(:each) do
+        allow_any_instance_of(certbase).to receive(:get_cert_pfx).and_return("")
+      end
+      it "returns nil" do
+        store = certstore.open(store_name)
+        cert_obj = store.get_pfx(thumbprint, store_location: CERT_SYSTEM_STORE_LOCAL_MACHINE, export_password: password)
+        expect(cert_obj).to eql("")
+      end
+    end
+
+    context "When passing valid thumbprint" do
+      let(:store_name) { "root" }
+      let(:password) { "1234" }
+      let(:thumbprint) { "d77803da081a5a556ab44c9cc74818767782c84b" }
+      let(:cert_pfx) { File.binread('.\spec\win32\assets\steveb.pfx') }
+      before(:each) do
+        allow_any_instance_of(certbase).to receive(:get_cert_pfx).and_return(cert_pfx)
```

Comment:
this seems similar (though inverse) to the above where even an invalid thumbprint would yield the same test result.